[3276] Fix incorrect template rendering

msopacua opened the pull request:
## Description

See commit messages.

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* #3276 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

msopacua committed:
test(3276): Add test for views

- Add a basic test to see if graphql template is rendering
- Add test for the failed rendering of the setting when overriding the
  template

msopacua committed:
fix(3276): Use get_template()

The `get_template()` call correctly fetches the template via the engine,
but to get the same template type as the one used in the other branch,
we need to access its template attribute, where the original template is
stored.

The engine template wrap the base template to allow direct
rendering by TemplateResponse, however this conflicts with the "template
via static files" approach taken by Strawberry.

Since I'm unfamiliar with the reasoning for this approach, we'll leave
it in tact.

msopacua committed:
Reuse test logic

Test is identical, but we need the decorator on one of them.

msopacua committed:
Add release file

patrick91 commented on RELEASE.md:
```diff
@@ -0,0 +1,5 @@
+Release type: patch
+
+Fix template rendering for the GraphQLView, so that variables are not consumed
+too early if one overrides the template using the well-known path. The symptom
+is that nothing loads because `JSON.parse("")` is a syntax error.
```
```suggestion
This release addresses an issue in Django's GraphQLView. Previously, variables were consumed too early when overriding the template, leading to a syntax error in the JavaScript code.
```

I think this reads a bit better 😊

patrick91 reviewed (approved):
looks great! thanks 😊

patrick91 committed:
Update RELEASE.md

patrick91 commented:
@msopacua looks like tests on older versions of django are failing 😊

msopacua commented:
Ah forgot about that. Django < 4.2 doesn't rewrite headers into WSGI environment variables yet.

msopacua committed:
compat(3276): Be compatible with older Django

Older Django versions, don't support requests-like headers for the test
client.

patrick91 commented:
I think this is not needed anymore after #3660
